core/src/main/java/brooklyn/location/access/PortForwardManager.java

```diff
@@ -92,6 +129,12 @@
      * subsequently be looked up using {@link #lookup(String, int)}.
      */
     public void associate(String publicIpId, HostAndPort publicEndpoint, int privatePort);
+
+    @Beta
+    public void addAssociationListener(AssociationListener listener, Predicate<? super AssociationMetadata> filter);
```

Comment:
Worth javadoc to say `Registers a listener, which will be notified each time a new port mapping is associated. See {@link #associate(String, HostAndPort, int)} and {@link #associate(String, HostAndPort, Location, int)}.`